Exiled.API/Features/Ragdoll.cs

```diff
@@ -268,43 +233,122 @@ public Vector3 Scale
         /// </summary>
         internal static HashSet<BasicRagdoll> IgnoredRagdolls { get; set; } = new();
 
+        /// <summary>
+        /// Creates a new ragdoll.
+        /// </summary>
+        /// <param name="networkInfo">The data associated with the ragdoll.</param>
+        /// <returns>The ragdoll.</returns>
+        /// <exception cref="ArgumentException">Provided RoleType is not a valid ragdoll role (Spectator, Scp079, etc).</exception>
+        /// <exception cref="InvalidOperationException">Unable to create a ragdoll.</exception>
+        public static Ragdoll Create(RagdollData networkInfo)
+        {
+            if (networkInfo.RoleType.GetRoleBase() is not IRagdollRole ragdollRole)
+                throw new ArgumentException($"Provided RoleType '{networkInfo.RoleType}' is not a valid ragdoll role.");
+
+            GameObject modelRagdoll = ragdollRole.Ragdoll.gameObject;
+
+            if (modelRagdoll == null || !Object.Instantiate(modelRagdoll).TryGetComponent(out BasicRagdoll ragdoll))
+                throw new InvalidOperationException("Unable to create a ragdoll.");
+
+            ragdoll.NetworkInfo = networkInfo;
+
+            Ragdoll doll = new(ragdoll)
+            {
+                Position = networkInfo.StartPosition,
+                Rotation = networkInfo.StartRotation,
+            };
+
+            return doll;
+        }
+
+        /// <summary>
+        /// Creates a new ragdoll.
+        /// </summary>
+        /// <param name="roleType">The <see cref="RoleTypeId"/> of the ragdoll.</param>
+        /// <param name="name">The name of the ragdoll.</param>
+        /// <param name="damageHandler">The damage handler responsible for the ragdoll's death.</param>
+        /// <param name="owner">The optional owner of the ragdoll.</param>
+        /// <returns>The ragdoll.</returns>
+        public static Ragdoll Create(RoleTypeId roleType, string name, DamageHandlerBase damageHandler, Player owner = null)
+            => Create(new(owner?.ReferenceHub ?? Server.Host.ReferenceHub, damageHandler, roleType, default, default, name, NetworkTime.time));
+
+        /// <summary>
+        /// Creates a new ragdoll.
+        /// </summary>
+        /// <param name="roleType">The <see cref="RoleTypeId"/> of the ragdoll.</param>
+        /// <param name="name">The name of the ragdoll.</param>
+        /// <param name="deathReason">The reason the ragdoll died.</param>
+        /// <param name="owner">The optional owner of the ragdoll.</param>
+        /// <returns>The ragdoll.</returns>
+        public static Ragdoll Create(RoleTypeId roleType, string name, string deathReason, Player owner = null)
+            => Create(new(owner?.ReferenceHub ?? Server.Host.ReferenceHub, new CustomReasonDamageHandler(deathReason), roleType, default, default, name, NetworkTime.time));
```

Comment:
```suggestion
        public static Ragdoll Create(RoleTypeId roleType, string name, string deathReason, Player owner = null)
            => Create(roleType, name, new CustomReasonDamageHandler(deathReason), owner);
```
We should call the other method to avoid repeating the same code.